File: content/snippets/cpp/language_basics_enums_and_constants.cpp
#include <iostream>
#include <string>

enum class Status {
    Pending,
    Running,
    Done,
};

constexpr int kMaxRetries = 3;

std::string to_string(Status status) {
    switch (status) {
        case Status::Pending:
            return "pending";
        case Status::Running:
            return "running";
        case Status::Done:
            return "done";
    }
    return "unknown";
}

int main() {
    Status status = Status::Running;
    std::cout << "status: " << to_string(status) << "\n";
    std::cout << "max retries: " << kMaxRetries << "\n";
    return 0;
}

File: content/snippets/go/language_basics_enums_and_constants.go
package main

import "fmt"

type Status int

const (
	Pending Status = iota
	Running
	Done
)

const MaxRetries = 3

func (status Status) String() string {
	switch status {
	case Pending:
		return "pending"
	case Running:
		return "running"
	case Done:
		return "done"
	default:
		return "unknown"
	}
}

func main() {
	status := Running
	fmt.Println("status:", status.String())
	fmt.Println("max retries:", MaxRetries)
}
